File: PythonConversion/functionLoadSharedFiles.py
import gurobipy as gp
import networkx as nx
import pandas as pd
from datetime import datetime

myRun = datetime.now().strftime("%H:%M:%S")
gurobi_env = gp.Env()

# Define your global variables as needed
edge = None
cL_orig = None
cU_orig = None
Len = None
c_orig = None
yy = None
SP_init = None
p = None
g = None
h = None
origin = None
destination = None
last_node = None
all_nodes = None
M_orig = None
delta1 = None
delta2 = None
b = None
last_node = None

numNodes = str(ARGS[1])  # Replace ARGS[1] with your desired argument values
density = str(ARGS[2])
Ins = str(ARGS[3])
dataSet = "N" + numNodes + "_d" + density

myFile = "./PrelimTestInstances/" + dataSet + "/" + dataSet + "_Ins_" + Ins + ".jl"
print("Ins", dataSet, "_", Ins, " Running...", myRun)

# Include your Julia files as needed
# You can translate the include statements into Python scripts as required.

epsilon = 1e-4
gurobi_env.setParam("OutputFlag", 0)

File: functionLoadSharedFiles.jl
using JuMP 
using Gurobi
using LightGraphs
using DataFrames, Query
using CSV
using TimerOutputs
using Dates
# using Polynomials #Be careful using Polynomials and JuMP pkg in Julia 1.9. They both have @variables so not preferred. 

myRun = Dates.format(now(), "HH:MM:SS")
global gurobi_env = Gurobi.Env()
global edge, cL_orig, cU_orig, Len, c_orig, yy, SP_init, p,g,h, origin, destination, last_node, all_nodes, M_orig, delta1, delta2, b, last_node
# global β # = rand(1:999)/1000
# gurobi_env.setParam("LogToConsole", 0)

to = TimerOutput()
global numNodes = string(ARGS[1])
global density = string(ARGS[2])
global Ins = string(ARGS[3])
# global myPath = "/Users/dinguyen/Library/CloudStorage/GoogleDrive-di.hoai.nguyen@gmail.com/Other\ computers/My\ Laptop/Documents/GitHub/Paper5"
global dataSet = "N"*string(numNodes)*"_d"*string(density)

myFile = "./PrelimTestInstances/"*dataSet*"/"*dataSet*"_Ins_"*Ins*".jl"
println("Ins ", dataSet,"_", Ins, " Running...", myRun)
include(myFile)
include("functionGbound.jl")
# println("Before function Hbound")
include("functionHbound.jl")

include("functionGetCellInfo.jl")
include("functionSelectArc.jl")
include("functionArcSplit.jl")
include("functionPartition.jl")
global epsilon = 1e-4
Gurobi.GRBsetintparam(gurobi_env, "OutputFlag", 0)